aspnetcore/blazor/fundamentals/dependency-injection.md

```diff
@@ -137,6 +137,21 @@ public class ComponentBase : IComponent
 }
 ```
 
+> [!NOTE]
+> Don't mark injected services as nullable. Instead, assign a default literal with the null-forgiving operator (`default!`). For example:
```

Comment:
I guess. One of the alternatives we were considering was to enable [constructor injection](https://github.com/dotnet/aspnetcore/issues/18088) by default in 7.0 which would remove some of this ugliness.

Reply:
Sorry for the nitpick, but I think it would be best to clarify as to why you shouldn't mark properties as nullable:

> **Since injected properties are always expected to be available, as a best practice** don't mark injected services as nullable. Instead...

The first time I read this, I thought that there might be some more technical reason as to why you shouldn't mark it as nullable.

Reply:
It's not just a best practice from the remarks made: It's how Blazor is designed to work. Therefore, an imperative remark should be fine.

Reply:
I see, so as a consumer, we shouldn't think of this as being "optional" at all. Thanks for the clarification!